client/block.c

```diff
@@ -839,7 +839,7 @@ void *add_block_callback(void *block, void *data)
 
 	pthread_mutex_unlock(&block_mutex);
 
-	if(res >= 0) {
+	if(res >= 0 && g_xdag_type == XDAG_POOL) {
```

Comment:
Blocks synchronization logic is initialized only for pools now, so xdag_sync_pop_block can be called only for pools